### PR TITLE
Update create/createFromImage to use new DataService.list

### PR DESCRIPTION
**Pull request by MarkDeMaria**
Make create/createFromImage promise compliant #799

### CODE REVIEW

File: app/scripts/controllers/create/createFromImage.js [hunk 1]
```diff
@@ -230,9 +230,9 @@ angular.module("openshiftConsole")
           $scope.memoryProblems = LimitRangesService.validatePodLimits($scope.limitRanges, 'memory', [$scope.container], project);
         };
 
-        DataService.list("limitranges", context, function(limitRanges) {
-          $scope.limitRanges = limitRanges.by("metadata.name");
-          if ($filter('hashSize')(limitRanges) !== 0) {
+        DataService.list("limitranges", context).then(function(resp) {
+          $scope.limitRanges = resp.by("metadata.name");
+          if ($filter('hashSize')(resp) !== 0) {
```
Comment (spadgett) [inline app/scripts/controllers/create/createFromImage.js]: This was existing code, but we shouldn't check `hashSize` against the responsive itself. Let's change this to

```js
if (!_.isEmpty($scope.limitRanges)) { ...
```

**Comment by MarkDeMaria** [comment]
@spadgett Oops! Slipped by. Changes made.

**Comment by spadgett** [comment]
[merge]

**Comment by spadgett** [comment]
Jasmine timeout

[merge]

**Pull request merged**